packages/console/src/components/TextInput/index.tsx

```diff
@@ -20,10 +29,22 @@ function TextInput(
     disabled,
     className,
     readOnly,
+    type = 'text',
     ...rest
   }: Props,
-  reference: ForwardedRef<HTMLInputElement>
+  reference: Ref<Nullable<HTMLInputElement>>
 ) {
+  const innerRef = useRef<HTMLInputElement>(null);
+  useImperativeHandle(reference, () => innerRef.current);
+
+  useEffect(() => {
+    if (type === 'number') {
+      innerRef.current?.addEventListener('wheel', (event) => {
```

Comment:
remove event listener on useEffect callback

Reply:
Also, why not directly set onWheel attr on the input field?

Reply:
This is because the `wheel` event listener is a passive event listener, so we cannot use `preventDefault` inside the listener.